packages/deep/src/index.ts

```diff
@@ -0,0 +1,63 @@
+import { Accessor, EffectFunction, NoInfer, OnEffectFunction, untrack } from "solid-js";
+
+/**
+ * createDeep - make a deep watch on the given object
+ * ```typescript
+ * export function deep<S, U>(
+ *   dep: Accessor<S>,
+ *   fn: (input: S, prevInput: S | undefined, prevValue: U | undefined) => U,
+ * ): (prevValue: U | undefined) => U;
+ * ```
+ * @param dep reactive dependency
+ * @param fn computation on input; the current previous content(s) of input and the previous value are given as arguments and it returns a new value
+ * @returns an effect function that is passed into createEffect. For example:
+ *
+ * ```typescript
+ * createEffect(
+ *     createDeep(
+ *         () => store,
+ *         () => { ... }
+ *     )
+ * );
+ * ```
+ */
+export function createDeep<S, Next extends Prev, Prev = Next>(
+  dep: Accessor<S>,
+  fn: OnEffectFunction<S, undefined | NoInfer<Prev>, Next>,
+): EffectFunction<undefined | NoInfer<Next>, NoInfer<Next>>;
+export function createDeep<S, Next extends Prev, Prev = Next>(
```

Comment:
What do you think about implementing it in a composable way?
I mean so that it could be used together with `on` (rather than instead of it), [`defer`](https://github.com/solidjs-community/solid-primitives/blob/main/packages/utils/src/index.ts#L162), or with any computation without a wrapper for stating explicit dependencies. I mean it's just a way of deep-tracking the store. Better to do as little as possible and let the user compose what he wants.

So `deep` or `trackDeep` may be a better name. `create___` would indicate that you are instantiating something reactive.

```ts
createEffect(() => {
  trackDeep(store)
  return untrack(() => { /*...*/ })
})

createEffect(on(
  [() => trackDeep(store), anotherSource],
  ([store, anotherValue]) => { /*...*/ },
  { defer: true }
))

import { defer } from "@solid-primitives/utils"
createEffect(defer(
  () => trackDeep(store),
  (store) => { /*...*/ },
))
```

Reply:
Loved the idea, much simpler and powerful implementation.